### PR TITLE
Feat / config deploy api

### PR DESCRIPTION
**Pull request by Spacelocust**
## Description

Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List any dependencies that are required for this change.

Closes #(issue) | No issue linked.

## Type of change

Please add any relevant information to the type of change you've made.

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [x] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
-   [ ] This change requires a documentation update (or is a documentation update)
-   [ ] HIGH IMPORTANCE (requires a fast merging)

## Has This Been Tested?

Please describe the tests that you ran to verify your changes. It is way more likely for a PR to be merged and approved with successful tests.

-   [x] Yes, I ran tests
-   [ ] No, I did not include any test
-   [ ] No test is needed for this change

### CODE REVIEW

File: .github/workflows/docker-api.yml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        with:
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_TOKEN }}
+
+      - name: Build and push Docker image
+        uses: docker/build-push-action@v5
+        with:
+          context: .
+          file: docker/prod/golang/Dockerfile
+
```
Comment (YummYume) [inline .github/workflows/docker-api.yml]:
```suggestion
```

File: .github/workflows/docker-api.yml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+          builder: ${{ steps.buildx.outputs.name }}
+          push: ${{ github.event_name != 'pull_request' }}
+          tags: spacelocust/fd-api:latest
+
```
Comment (YummYume) [inline .github/workflows/docker-api.yml]:
```suggestion
```

File: .github/workflows/docker-migrations.yml [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        with:
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_TOKEN }}
+
+      - name: Build and push Docker image
+        uses: docker/build-push-action@v5
+        with:
+          context: .
+          file: docker/prod/atlas/Dockerfile
+
```
Comment (YummYume) [inline .github/workflows/docker-migrations.yml]:
```suggestion
```

File: .github/workflows/docker-migrations.yml [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+          builder: ${{ steps.buildx.outputs.name }}
+          push: ${{ github.event_name != 'pull_request' }}
+          tags: spacelocust/fd-migrations:latest
+
```
Comment (YummYume) [inline .github/workflows/docker-migrations.yml]:
```suggestion
```

File: docker/prod/caddy/Caddyfile [hunk 5]
```diff
@@ -0,0 +1,8 @@
+{
+    email {$ACME_EMAIL}
+    admin off
+}
+
+api.yam-yam.dev {
+    reverse_proxy * api:5000
+}
```
Comment (YummYume) [inline docker/prod/caddy/Caddyfile]:
```suggestion
}

```

File: docker/prod/golang/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+  ca-certificates \
+  && rm -rf /var/lib/apt/lists/*
+
+# Copy the binary from the builder stage
+COPY --from=builder /go/src/app/fd-democracy /go/bin/fd-democracy
+
+ENV PATH="/go/bin:${PATH}"
+
+ENTRYPOINT ["fd-democracy", "gin-server"]
+
```
Comment (YummYume) [inline docker/prod/golang/Dockerfile]:
```suggestion
```

File: docker/prod/golang/Dockerfile [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+# Remove the migrations folder and build the binary
+RUN rm -rf /migrations && GOOS=linux GOARCH=amd64 go build -o fd-democracy main.go
+
+FROM debian:buster-slim AS runner
+
+RUN apt-get update && apt-get install -y \
+  curl \
+  ca-certificates \
+  && rm -rf /var/lib/apt/lists/*
```
Comment (YummYume) [inline docker/prod/golang/Dockerfile:21]: Je pense qu'il y a plus besoin mais pg

Comment (Spacelocust) [reply docker/prod/golang/Dockerfile:21]: Je l'ai mis, car j'ai eu une erreur lors de mon test de l'image après build

**Review by YummYume — APPROVED** [review]

**Pull request merged**